api/src/services/attachment-service.ts

```diff
@@ -828,4 +848,112 @@ export class AttachmentService extends DBService {
 
     return { ...attachmentResult, key };
   }
+
+  /**
+   * Handle deletion of Project Attachment.
+   * If attachment is a report, delete authors
+   * Otherwise, delete the publish record and the attachment.
+   * If attachment was published, delete the publish record and delete from Platform.
```

Comment:
This doc is a bit confusing. "If report, delete authors, otherwise delete publish record and attachment". We want to delete the publish record in both cases regardless of attachment type.

One way to write this out is just to repeat yourself for each type:

```
If report:
 - Delete stuff
 - Delete authors
 - Delete publish record
if not report:
 - Delete stuff
 - Delete authors
 - Delete publish record
```